### ISSUE TITLE
printNcML create 0-size variables with undefined type

### ISSUE THREAD

**Issue by heikoklein** [metno/fimex#26]
Hi,

printNcML on 0-size variables, i.e. projections give type="NAT" when read from grib-files. NAT is a undefined type in the ncml-schema: https://www.unidata.ucar.edu/schemas/netcdf/ncml-2.2.xsd

```xml
<variable name="projection_regular_ll" type="NAT" >
  <attribute name="grid_mapping_name" type="string" value="latitude_longitude" />
  <attribute name="earth_radius" type="double" value="6367470" />
  <attribute name="proj4" type="string" value="+proj=longlat +a=6367470 +e=0 +no_defs" />
</variable>
```

type should be omitted here, or set to something valid (it doesn't matter since size=0). When converting to netcdf-files, the NAT-type is translated to 'int':
```xml
<variable name="projection_regular_ll" type="int" >
  <attribute name="grid_mapping_name" type="string" value="latitude_longitude" />
  <attribute name="earth_radius" type="double" value="6367470" />
  <attribute name="proj4" type="string" value="+proj=longlat +a=6367470 +e=0 +no_defs" />
</variable>
```

**Comment by heikoklein** [comment]
This case was tested with fimex 1.5 and 1.5.2.

**Comment by alexander-buerger-met-no** [comment]
Do you have an example for this?

The output from `fimex --input.file test/twoCoordsTest.nc --input.printNcML` includes

    <variable name="projection_1" type="int" >
      <attribute name="proj4" type="string" value="+proj=stere +lat_0=90 +lon_0=0 +lat_ts=60 +units=m +a=6.371e+06 +e=0 +no_defs" />
      
    </variable>

It looks like `projection_1` has type `int` in both netcdf and ncml.

**Comment by heikoklein** [comment]
As I wrote "when reading from grib-files", any grib-file will do I think.

**Comment by alexander-buerger-met-no** [comment]
Sorry, I missed "from grib files".

The origin of the problem seems to be [GribCDMReader.cc:1007](https://github.com/metno/fimex/blob/a356f606713dd233b08a4eb437581980de8fcc63/src/GribCDMReader.cc#L1007) and similar code in other classes.

**Comment by heikoklein** [comment]
The problem is in the ncml-writer. Fimex has a type 'NAT' which is fine. ncml does not have a type NAT, so a translation is required.

A minimal example:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<netcdf xmlns="http://www.unidata.ucar.edu/namespaces/netcdf/ncml-2.2"
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="http://www.unidata.ucar.edu/namespaces/netcdf/ncml-2.2 http://www.unidata.ucar.edu/schemas/netcdf/ncml-2.2.xsd">
<variable name="x"  />
</netcdf>
```
translates with `fimex --input.file=notype.ncml --input.printNcML` to
```xml
<variable name="x" type="NAT" />
```

**Comment by alexander-buerger-met-no** [comment]
Should be closed with commit c905c414b2.
